demo/src/client/FileAttachmentPage/FileDisplayPanel.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+import { Panel } from 'react-bootstrap';
+import { MY_ATTACHMENTS_DIR } from "./constants";
+import { Draft, produce } from "immer";
+import { ActionURL } from "@labkey/api";
+import { FileAttachmentModel, SavedFileModel } from "./models";
+import { getUploadedFiles } from "./actions";
+
+export const App : FC = memo(() => {
+
+    const [fileAttachmentModel, setFileAttachmentModel] = useState<FileAttachmentModel>(new FileAttachmentModel());
+
+    //equivalent to componentDidMount and componentDidUpdate
+    useEffect(() => {
+        getUploadedFiles(ActionURL.getContainer(), MY_ATTACHMENTS_DIR, false).then((files:Array<SavedFileModel>) => {
+            if (files?.length > 0) {
+                const updatedModel = produce(fileAttachmentModel, (draft: Draft<FileAttachmentModel>) => {
+                    draft['savedFiles'] = files;
+                });
+                setFileAttachmentModel(updatedModel);
+            }
+            else {
+                setFileAttachmentModel(new FileAttachmentModel());
+            }
+        });
+    }, []);
+
+    const showImg = useCallback((evt: any) => {
```

Comment:
See comment below. I don't think this is needed if you just use the `savedFile.href` directly in the `<a>` tag